react-hot-button/my-app/src/button.tsx

```diff
@@ -0,0 +1,32 @@
+import { useState } from 'react';
+import './button.css';
+
+export function Button() {
+  const [index, setIndex] = useState(0);
+  let color = 'purple';
+
+  if (index <= 3) {
+    color = 'purple';
+  } else if (index <= 6) {
+    color = 'lightPurple';
+  } else if (index <= 9) {
+    color = 'red';
+  } else if (index <= 12) {
+    color = 'orange';
+  } else if (index <= 15) {
+    color = 'yellow';
+  } else {
+    color = 'purple';
+  }
+  const handleClick = () => {
+    setIndex(() => index + 1);
```

Comment:
The code would look a lot nicer if you just call it without using an arrow function:
`setIndex(index + 1);`